src/EditorFeatures/Core/ValueTracking/ValueTrackingService.Visitor.cs

```diff
@@ -0,0 +1,232 @@
+                        IFieldReferenceOperation fieldReferenceOperation => AddOperationAsync(operation, fieldReferenceOperation.Member, cancellationToken),
+                        IPropertyReferenceOperation propertyReferenceOperation => AddOperationAsync(operation, propertyReferenceOperation.Member, cancellationToken),
+                        ILocalReferenceOperation localReferenceOperation => AddOperationAsync(operation, localReferenceOperation.Local, cancellationToken),
+                        _ => Task.CompletedTask
+                    };
+            }
+
+            private static bool IsArgument(IOperation? operation, [NotNullWhen(returnValue: true)] out IArgumentOperation? argumentOperation)
+            {
+                while (operation is not null)
+                {
+                    if (operation is IArgumentOperation tmpArgumentOperation)
+                    {
+                        argumentOperation = tmpArgumentOperation;
+                        return true;
+                    }
+
+                    operation = operation.Parent;
+                }
+
+                argumentOperation = null;
+                return false;
+            }
+
+            private static bool IsReturn(IOperation? operation)
+            {
+                while (operation is not null)
+                {
+                    if (operation is IReturnOperation)
+                    {
+                        return true;
+                    }
+
+                    operation = operation.Parent;
+                }
+
+                return false;
+            }
+
+            private async Task VisitLiteralAsync(ILiteralOperation literalOperation, CancellationToken cancellationToken)
+            {
+                if (literalOperation.Type is null)
+                {
+                    return;
+                }
+
+                await AddOperationAsync(literalOperation, literalOperation.Type, cancellationToken).ConfigureAwait(false);
+            }
+
+            private async Task VisitReturnAsync(IReturnOperation returnOperation, CancellationToken cancellationToken)
+            {
+                if (returnOperation.ReturnedValue is null)
+                {
+                    return;
+                }
+
+                await VisitAsync(returnOperation.ReturnedValue, cancellationToken).ConfigureAwait(false);
+            }
+
+            private async Task AddOperationAsync(IOperation operation, ISymbol symbol, CancellationToken cancellationToken)
+            {
+                _ = await ProgressCollector.TryReportAsync(
+                        Solution,
+                        operation.Syntax.GetLocation(),
+                        symbol,
+                        cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            private async Task TrackArgumentsAsync(ImmutableArray<IArgumentOperation> argumentOperations, CancellationToken cancellationToken)
```

Comment:
aside 1: i'm surprised all this code is living in EditorFeatures.
aside 2: where does this code run? i would expect in OOP.

Reply:
1: Where do you expect it live? 
2: I haven't moved this to OOP yet. The FAR implementation is in OOP, which handles the expensive part for now.

Reply:
yeah.  this should move to oop eventually.
here's why:

you'll get the results back from OOP, but then you start processing them here.  thi smeans that VS will now have to build compilations/semanticmodels/iop-trees for lots of data in proc that it won't have.  Conversly though, OOP will ahve much of that data, as it just computed it :)

Reply:
https://github.com/dotnet/roslyn/issues/52435 to do OOP